sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/Metadata.java

```diff
@@ -239,6 +239,24 @@ public interface Metadata extends Iterable<Metadata.MetadataEntry> {
    */
   Principals principals();
 
+  /**
+   * Add an HTTP response code to this metadata.
+   * This will only take effect when HTTP transcoding is in use. It will be ignored for gRPC requests.
+   *
+   * @param code The success status code to add.
+   * @return a copy of this metadata with the HTTP response code set.
+   */
+  Metadata withHttpResponseCode(StatusCode.Success code);
```

Comment:
How about adding some integration tests?

Reply:
of course, will update. tks